lib/persistence.js

```diff
@@ -162,18 +162,31 @@ Persistence.readHashAll = function(hash, callback) {
 };
 
 // Return the value associated with the key *key* (no associated hash)
-Persistence.readKey = function (key, callback) {
+Persistence.readKey = function (key, callback, expireTTL) {
+  var multi = Persistence.redis().multi();
+
   logging.debug('readKey:', key);
-  Persistence.redis().get(key, function (err, reply) {
-    if(err) throw new Error(err);
+  multi.get(key, function (err, reply) {
+    if (err) throw new Error(err);
+    if (expireTTL) {
+      multi.expire(key, expireTTL);
+    }
```

Comment:
To be clear: The callback for get, will be called after we exec. So, this expire will never be called.